src/vs/workbench/contrib/debug/browser/breakpointsView.ts

```diff
@@ -739,6 +763,19 @@ class DataBreakpointsRenderer implements IListRenderer<DataBreakpoint, IDataBrea
 		} else {
 			data.accessType.textContent = '';
 		}
+		if (dataBreakpoint.condition && dataBreakpoint.hitCondition) {
+			data.condition.textContent = localize('expressionAndHitCount', "Condition: {0} | Hit Count: {1}", dataBreakpoint.condition, dataBreakpoint.hitCondition);
+		} else {
+			data.condition.textContent = dataBreakpoint.condition || dataBreakpoint.hitCondition || '';
+		}
+
+		const primary: IAction[] = [];
+		this.breakpointSupportsCondition.set(!session || !!session.capabilities.supportsConditionalBreakpoints);
+		this.breakpointItemType.set('dataBreakpoint');
```

Comment:
The usual way to do this would be with a scoped context key service, which avoids having to re-evaluate context keys globally in vs code.

Reply:
Yeah, but I think it's been wrong for a long time for other breakpoint types

Reply:
Thanks for the feedback. This code currently replicates existing code for function breakpoints.